src/modelbench/utilities.py

```diff
@@ -8,3 +10,26 @@ def group_by_key(items: Sequence, key=Callable) -> Mapping:
     for item in items:
         groups[key(item)].append(item)
     return groups
+
+
+class ProgressTracker:
+    def __init__(self, total, print_updates=False, shared_count=None, lock=None):
+        if shared_count is None:
+            self.complete_count = Value(ctypes.c_double, 0.0)
+        else:
+            self.complete_count = shared_count
+        self.total = total
+        self.print_updates = print_updates
+        self.lock = lock
+
+    def increment(self):
+        if self.lock:
+            with self.lock:
+                self._increment()
+        else:
+            self._increment()
+
+    def _increment(self):
+        self.complete_count.value += 1.0
+        if self.print_updates:
+            print({"progress": self.complete_count.value / self.total})
```

Comment:
This will crash if `self.total` is 0. Which is unlikely. But IME anything that can go wrong will, given enough opportunities, so I'd say any one of these options could work:

1. test `if self.total` before doing the division
2. initialize it in the constructor
3. make `self.total` a `property` with a `setter` that rejects zero